listenbrainz/webserver/views/stats_api.py

```diff
@@ -477,7 +474,23 @@ def get_daily_activity(user_name: str):
     if stats is None or getattr(stats, stats_range) is None:
         raise APINoContent('')
 
-    daily_activity = [x.dict() for x in getattr(stats, stats_range).daily_activity]
+    daily_activity_unprocessed = [x.dict() for x in getattr(stats, stats_range).daily_activity]
+    daily_activity = {calendar.day_name[day]: [{"hour": hour, "listen_count": 0} for hour in range(0, 24)] for day in range(0, 7)}
+
+    for day, day_data in daily_activity.items():
+        for hour_data in day_data:
+            hour = hour_data["hour"]
+
+            found = False
+            for entry in daily_activity_unprocessed:
```

Comment:
you can use a `for...else` here to get rid of the `found` variable. (https://book.pythontips.com/en/latest/for_-_else.html)

Reply:
Neat trick!